### PR TITLE
Maya light probe export

### PR DESCRIPTION
**Pull request by jerome-jacob**
This will produce:
```xml
<COLLADA …>
  <extra>
    <technique profile="OpenCOLLADAMaya">
      <library_light_probes>
        <light_probe id="_7252" name="LightProbe1"/>
      </library_light_probes>
    </technique>
  </extra>

<node…>
  <matrix>...</matrix>
  <extra>
    <technique profile=”OpenCOLLADAMaya”>
      <instance_light_probe url=”#_7252”/>
    </technique>
  </extra>
</node>
```

### CODE REVIEW

File: COLLADAMaya/include/COLLADAMayaLightProbeExporter.h [hunk 1]
```diff
@@ -0,0 +1,188 @@
+/*
+    Copyright (c) 2008-2009 NetAllied Systems GmbH
```
Comment (RemiArnaud) [inline COLLADAMaya/include/COLLADAMayaLightProbeExporter.h]: please fix copyright

Copyright (c) 2015 Starbreeze AB

are some portions of the code below from Feeling Software ? Sony? Alias?

File: COLLADAMaya/include/COLLADAMayaSyntax.h [hunk 2]
```diff
@@ -36,7 +36,7 @@ namespace COLLADAMaya
 
     /** Profile specific extra attribute parameters. */
     static const char* PROFILE_COLLADA                              = "COLLADA";
-    static const char* PROFILE_MAYA                                 = "OpenCOLLADAMaya";
+    static const String PROFILE_MAYA                                 = "OpenCOLLADAMaya";
     static const char* PROFILE_MAX                                  = "OpenCOLLADA3dsMax";
```
Comment (RemiArnaud) [inline COLLADAMaya/include/COLLADAMayaSyntax.h]: Why is only PROFILE_MAYA switched to String ? Why not switching all of them, or not changing PROFILE_MAYA definition?

File: COLLADAMaya/src/COLLADAMayaLightProbeExporter.cpp [hunk 3]
```diff
@@ -0,0 +1,243 @@
+/*
```
Comment (RemiArnaud) [inline COLLADAMaya/src/COLLADAMayaLightProbeExporter.cpp]: please fix copyright

File: COLLADAMaya/src/COLLADAMayaSceneElement.cpp [hunk 4]
```diff
@@ -37,6 +37,7 @@ namespace COLLADAMaya
             , mIsForced ( false )
             , mIsVisible ( false )
             , mHasJoint ( false )
+            , mIsPhysicNode(false)
```
Comment (RemiArnaud) [inline COLLADAMaya/src/COLLADAMayaSceneElement.cpp]: is this intended?

File: COLLADAStreamWriter/src/COLLADASWLibrary.cpp [hunk 5]
```diff
@@ -43,5 +45,30 @@ namespace COLLADASW
...
+    void Library::closeExtraLibrary()
+    {
+        if (mLibraryOpen)
+        {
+            mLibraryCloser.close();
+            mTechnique.closeTechnique();
+            mExtra.closeExtra();
+        }
+
+        mLibraryOpen = false;
```
Comment (RemiArnaud) [inline COLLADAStreamWriter/src/COLLADASWLibrary.cpp]: in this function, libraryOpen=false is done outside of the if() statement
in the function above, libraryOpen=true is done inside the if() statement.

Probably you want to make this line inside the if() statement in both functions?

File: COLLADAMaya/include/COLLADAMayaLightProbeExporter.h (hunk 1 above)
Comment (jerome-jacob) [inline COLLADAMaya/include/COLLADAMayaLightProbeExporter.h]: Woops.... Copy paste...

File: COLLADAMaya/include/COLLADAMayaSyntax.h (hunk 2 above)
Comment (jerome-jacob) [inline COLLADAMaya/include/COLLADAMayaSyntax.h]: I needed PROFILE_MAYA to be a String. I'll update the others later in another PR.

File: COLLADAMaya/src/COLLADAMayaSceneElement.cpp (hunk 4 above)
Comment (jerome-jacob) [inline COLLADAMaya/src/COLLADAMayaSceneElement.cpp]: Uninitialized variable. Should have been in a separate PR.

File: COLLADAStreamWriter/src/COLLADASWLibrary.cpp (hunk 5 above)
Comment (jerome-jacob) [inline COLLADAStreamWriter/src/COLLADASWLibrary.cpp]: Good catch :)

**Pull request merged**